### ISSUE TITLE
[0.7.0] Error: Constructors have to be declared either "public" or "internal"

### ISSUE THREAD

**Issue by yosriady** [trufflesuite/truffle#3260]
- [ ] I've asked for help in the [Truffle Gitter](http://gitter.im/Consensys/truffle) before filing this issue.

---------------------------

## Issue

Truffle warns on missing constructor visibility, even though it's optional on Solidity 0.7.0.

## Steps to Reproduce

Clone https://github.com/yosriady/testing-solidity and run `yarn test`

## Expected Behavior

No warnings should be displayed.

## Actual Results

Truffle raises the following warnings on the constructor visibility:

```
>>>>> WARNING <<<<<<
Failed to parse file: "./contracts/Counter.sol". No data will collected for its contract(s).
NB: some Solidity 0.6.x syntax is not supported by the JS parser yet.
Please report the error below to github.com/consensys/solidity-parser-antlr
>>>>>>>>>>>>>>>>>>>>
Error: Constructors have to be declared either "public" or "internal"
```

## Environment

OSX 10.15.5
Truffle v5.1.29 (core: 5.1.29)
Solidity - 0.7.0 (solc-js)
Node v13.7.0
Web3.js v1.2.1

**Comment by haltman-at** [comment]
While there is a little extra processing for things like imports, for the most part Truffle just defers to Solidity for compilation; it doesn't generate any warnings or errors there on its own.  In this case, I see you're using Solidity 0.6.10, so you're getting Solidity 0.6.10 errors and warnings.  If you want the behavior of Solidity 0.7.0, you'll have to use Solidity 0.7.0.  Or is there something I'm missing here, or some reason that this should behave differently?  This all seems correct to me, but maybe I'm missing something.  Thanks!

**Comment by gnidan** [comment]
Looks like this might be an error coming from [solhint](https://github.com/protofire/solhint). Looks like your project is on `^3.0.0`, which might not mean you have the latest v3.2.0? In any event, this doesn't seem like a Truffle error. We'll try to reproduce this ourselves to make sure, but in the meantime, let us know if upgrading your dependencies fixes this!

**Comment by fvictorio** [comment]
FYI this error comes from `eth-gas-reporter` ([line](https://github.com/cgewecke/eth-gas-reporter/blob/0714dd1d76b7d420942d29f83c569fc82a5704f7/lib/utils.js#L133)).

**Comment by haltman-at** [comment]
OK, since the issue seems to be coming from `eth-gas-reporter`, we're going to close this; you may want to report it to them, however.  Thanks to @fvictorio for finding the problem!
